### PR TITLE
JonathanCrd - 5797 Fixing ResourceType value

### PR DESCRIPTION
**Pull request by JonathanCrd**
# Description
<i>Add your description here!</i>
ADO item 5797.

Fixing the `ResourceType` value emitted from ResourceOperationWriter.cs to the correct one.

# Checklist

To ensure a quick review and merge, please ensure:
- [X] The PR has a understandable title and description explaining the _why_ and _what_.
- [X] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [X] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ internal class ResourceOperation : TypeProvider
         private BuildContext<MgmtOutputLibrary> _context;
         private PagingMethod[]? _pagingMethods;
 
-        protected OperationGroup _operationGroup;
+        public OperationGroup _operationGroup { get; protected set; }
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs:28]: We have added internal getter for OperationGroup in [Resource class](https://github.com/Azure/autorest.csharp/blob/f1fe1343f2a000fe3ed1d7fd7147d9643dd70f98/src/AutoRest.CSharp/Mgmt/Output/Resource.cs#L22). Can we do the same here?

Comment (ShivangiReja) [reply src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs:28]: nit: `_operationGroup` -> `OperationGroup`

File: samples/Azure.Management.Storage/Generated/BlobContainerOperations.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ protected BlobContainerOperations()
         {
         }
 
-        public static readonly ResourceType ResourceType = "Azure.Management.Storage/BlobContainerOperations";
+        public static readonly ResourceType ResourceType = "Microsoft.Storage/storageAccounts/blobServices/default/containers";
```
Comment (ShivangiReja) [inline samples/Azure.Management.Storage/Generated/BlobContainerOperations.cs:24]: Can we add a test to verify the `ResourceType` value?

Comment (m-nash) [reply samples/Azure.Management.Storage/Generated/BlobContainerOperations.cs:24]: we have unit tests here which is where that value is coming from https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/Mgmt/ProviderSegmentDetectionTests.cs

**Comment by JonathanCrd** [comment]
@ShivangiReja, I made changes based on your comments.

**Review by ShivangiReja — APPROVED** [review]

**Review by m-nash — APPROVED** [review]

**Pull request merged**